h2o-r/h2o-package/R/explain.R

```diff
@@ -1,5 +1,31 @@
 ########################################### UTILS ##################################################
 
+#' Works like match.arg but ignores case
+#' @param arg argument to match that should be declared as a character vector containing possible values
+#' @param choices argument to choose from (OPTIONAL)
+#' @return matched arg
+case_insensitive_match_arg <- function(arg, choices) {
+  var_name <- as.character(substitute(arg))
+  if (missing(choices))
+    choices <- eval(formals(sys.function(-1))[[var_name]])
+  orig_choices <- choices
+
+  if (identical(arg, eval(formals(sys.function(-1))[[var_name]])))
+    arg <- choices[[1]]
+
+  choices <- tolower(choices)
+
+  if (length(arg) != 1)
+    stop(sprintf("'%s' must be of length 1", var_name), call. = FALSE)
+
+  arg <- tolower(arg)
+
+  i <- pmatch(arg, choices, nomatch = 0L, duplicates.ok = FALSE)
+  if (all(i == 0L) || length(i) != 1)
+    stop(sprintf("'%s' should be one of %s", var_name, paste(dQuote(orig_choices), collapse = ", ")), call. = FALSE)
+  return(orig_choices[[i]])
+}
```

Comment:
looks like a global utility function, can't it go to `zzz.R` (as it looks like the internal utility module)

Reply:
The `zzz.R` has a very specific purpose during loading the R package. I'd rather create something like `utils.R` someday in near future in a separate PR.

```
If you use .onLoad(), consider using .onUnload() to clean up any side effects. By convention, .onLoad() and friends are usually saved in a file called R/zzz.R. (Note that .First.lib() and .Last.lib() are old versions of .onLoad() and .onUnload() and should no longer be used.)
```
(https://r-pkgs.org/Code.html#when-you-do-need-side-effects)